docs/datafetching.md

```diff
@@ -168,12 +167,34 @@ enum ShowGenre {
 
 We can write a datafetcher with the following signature:
 ```java
-@DgsData(parentType = "Query", field = "shows")
+@DgsQuery
 public List<Show> shows(@InputArgument String title, @InputArgument ShowFilter filter)
 ```
 
+The `@InputArgument` annotation will use the name of the method argument to match it with the name of an input argument sent in the query.
 Optionally we can specify the `name` argument in the `@InputArgument` annotation, if the argument name doesn't match the method argument name.
 
+The framework converts input arguments to Java/Kotlin types.
+The first step for converting input arguments is `graphql-java` using scalar implementations to convert raw string input into whatever type the scalar represents.
+A GraphQL `Int` becomes an `Integer` in Java, a formatted date string becomes a `LocalDateTime` (depending on the scalars you're using!), lists become an instance of `java.util.ArrayList`.
+Input objects are represented as a `Map<String, Object>` in `graphql-java`.
+
+The next step is the DGS Framework converting the `Map<String, Object>` to the Java/Kotlin classes that you use for the `@InputArgument`.
+For Java classes, the framework creates a new instance of the class using the no-arg constructor.
+This implies that a no-arg constructor is required.
+It then sets each field of the instance to the input argument values.
+
+For Kotlin Data classes, the instance can only be created by passing in all arguments in the constructor.
+This means you have to make sure to make fields optional in the data class when the fields are optional in the GraphQL schema!
+
+If you're using the [Codegen](generating-code-from-schema) plugin (you really should!), the input types will work perfectly out of the box.
```

Comment:
Should we add a note about adding type mappings for scalars in codegen to ensure they are correctly handled?

Reply:
good point. Added.